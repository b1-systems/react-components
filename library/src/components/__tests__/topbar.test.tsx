import HelpOutlineIcon from "@mui/icons-material/HelpOutline";
import { IconButton } from "@mui/material";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import TopBar, { NotificationHistory } from "../topbar";

it("should render consistently with minimal arguments", () => {
  const tree = renderer.create(<TopBar />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("Correct functionality", () => {
  const simplestTopbar = (
    <TopBar applicationTitle="My Test Application" helpLink="https://example.com" />
  );
  const simplestTopbarShallow = shallow(simplestTopbar);
  it("should show the passed application title", () => {
    expect(simplestTopbarShallow.text()).toEqual("My Test Application");
  });
  it("Should show no notificationHistory if not requested", () => {
    expect(
      simplestTopbarShallow.containsMatchingElement(
        <NotificationHistory
          pastNotifications=""
          noNotificationsYet=""
          createdAtFormat={(_) => ""}
        />,
      ),
    ).toEqual(false);
  });
  it("Should show a help icon if any helpLink is passed", () => {
    expect(
      simplestTopbarShallow.containsMatchingElement(
        <IconButton color="inherit" href="https://example.com" target="_blank">
          <HelpOutlineIcon />
        </IconButton>,
      ),
    ).toBeTruthy();
  });
});
