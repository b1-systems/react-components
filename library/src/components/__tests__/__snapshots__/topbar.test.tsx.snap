// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render consistently with minimal arguments 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-bcxrb8-MuiAppBar-root{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;width:100%;box-sizing:border-box;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;position:fixed;z-index:1100;top:0;left:auto;right:0;background-color:#1976d2;color:#fff;z-index:1201;}@media print{.css-bcxrb8-MuiAppBar-root{position:absolute;}}",
      }
    }
    data-emotion="css bcxrb8-MuiAppBar-root"
  />,
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-ivnu1y-MuiPaper-root-MuiAppBar-root{background-color:#fff;color:rgba(0, 0, 0, 0.87);-webkit-transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;box-shadow:0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;width:100%;box-sizing:border-box;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;position:fixed;z-index:1100;top:0;left:auto;right:0;background-color:#1976d2;color:#fff;z-index:1201;}@media print{.css-ivnu1y-MuiPaper-root-MuiAppBar-root{position:absolute;}}",
      }
    }
    data-emotion="css ivnu1y-MuiPaper-root-MuiAppBar-root"
  />,
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-ivnu1y-MuiPaper-root-MuiAppBar-root"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-1t29gy6-MuiToolbar-root{position:relative;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-left:16px;padding-right:16px;min-height:56px;}@media (min-width:600px){.css-1t29gy6-MuiToolbar-root{padding-left:24px;padding-right:24px;}}@media (min-width:0px) and (orientation: landscape){.css-1t29gy6-MuiToolbar-root{min-height:48px;}}@media (min-width:600px){.css-1t29gy6-MuiToolbar-root{min-height:64px;}}",
        }
      }
      data-emotion="css 1t29gy6-MuiToolbar-root"
    />
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
    />
  </header>,
]
`;
